root/static/scripts/common/components/sidebar/AcousticBrainz.js

```diff
@@ -0,0 +1,97 @@
+/*
+ * @flow
+ * Copyright (C) 2020 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import * as React from 'react';
+
+import hydrate from '../../../../../utility/hydrate';
+import {SidebarProperty, SidebarProperties} from
+  '../../../../../layout/components/sidebar/SidebarProperties';
+
+const SidebarAcousticBrainz = ({recording}: {recording: RecordingT}) => {
+  const [count, setCount] = React.useState(0);
+  const [data, setData] = React.useState(null);
+
+  function fetchAcousticBrainzCount() {
+    const countUrl = `//acousticbrainz.org/api/v1/${recording.gid}/count`;
+    fetch(countUrl).then(
+      resp => resp.json(),
+    ).then(
+      data => {
+        setCount(data.count);
+        if (data.count) {
+          fetchAcousticBrainzData();
+        }
+      },
+    );
+  }
+
+  function fetchAcousticBrainzData() {
+    const dataUrl = '//acousticbrainz.org/api/v1/low-level?' +
+      `recording_ids=${recording.gid}&` +
+      'features=tonal.key_key;tonal.key_scale;tonal.key_strength;rhythm.bpm';
+    fetch(dataUrl).then(
+      resp => resp.json(),
+    ).then(
+      data => {
+        setData(data[recording.gid][0]);
+      },
+    );
+  }
+
+  function keyLabel(key) {
+    if (!key.includes('#')) {
+      return key;
+    }
+    return {
+      'A#': 'A♯/B♭',
+      'C#': 'C♯/D♭',
+      'D#': 'D♯/E♭',
+      'F#': 'F♯/G♭',
+      'G#': 'G♯/A♭',
```

Comment:
I'd probably move this to a `const KEY_LABELS` or something after the imports, just to avoid the extra allocation on each call.

Reply:
Sure, I'll fix that